### PR TITLE
Vault pallet - Added function to calculate shares from asset

### PR DESCRIPTION
**Pull request by Matteocolella1995**
## Issue
I needed a function to calculate the amount of `lp_tokens` corresponding to an amount of asset given the current vault conditions. 

## Description
Added a trait function `calculate_lp_tokens_from_asset_amount` and its pallet implementation.

### CODE REVIEW

**Review by kevin-composable — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

File: frame/vault/src/lib.rs [hunk 1]
```diff
@@ -849,6 +849,25 @@ pub mod pallet {
...
 
+		fn do_calculate_lp_tokens_from_asset_amount(
+			vault_id: &T::VaultId,
+			vault: &VaultInfo<T>,
+			asset_amount: T::Balance,
+		) -> Result<BalanceOf<T>, DispatchError> {
+			let total_lp_issuance = T::Currency::total_issuance(vault.lp_token_id);
+			let aum = Self::assets_under_management(vault_id)?;
+
+			let a = <T::Convert as Convert<T::Balance, u128>>::convert(asset_amount);
```
Comment (dzmitry-lahoda) [inline frame/vault/src/lib.rs:860]: so you have functions which receives 3 numbers, converts and converts back?

Comment (dzmitry-lahoda) [reply frame/vault/src/lib.rs:860]: which is same as on lines 81x

Comment (dzmitry-lahoda) [reply frame/vault/src/lib.rs:860]: can we manage some code reuse?

Comment (dzmitry-lahoda) [reply frame/vault/src/lib.rs:860]: in this pr

File: frame/composable-traits/src/vault.rs [hunk 2]
```diff
@@ -112,6 +112,11 @@ pub trait Vault {
 		vault_id: &Self::VaultId,
 		lp_amount: Self::Balance,
 	) -> Result<Self::Balance, DispatchError>;
+
+	fn calculate_lp_tokens_from_asset_amount(
```
Comment (dzmitry-lahoda) [inline frame/composable-traits/src/vault.rs:116]: in AMM DEX traits (pablo) they call it `amount_of_lp_token_for_added_liquidity`

File: frame/vault/src/lib.rs (hunk 1 above)
Comment (Matteocolella1995) [reply frame/vault/src/lib.rs]: You're right, I have refactored the code with your suggestions, let me know if it's better now or if I should change something else too

File: frame/composable-traits/src/vault.rs (hunk 2 above)
Comment (Matteocolella1995) [reply frame/composable-traits/src/vault.rs]: I have renamed the function in the last commit together with the refactor you suggested above

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**